### ISSUE TITLE
Add support for downloading resolved specs

### ISSUE THREAD

**Issue by olensmar** [swagger-api/swaggerhub-maven-plugin#44]
My OAS3 spec references a domain - when downloading I would like to get that resolved in the same was as the Export/Resolved menu action in SwaggerHub

**Comment by scorpioa** [comment]
I found workaround using swagger-codegen-maven-plugin with:
`<execution>
						<id>spec-generator</id>
						<goals>
							<goal>generate</goal>
						</goals>
						<configuration>
							<inputSpec>${swaggerhub.apis.host}/${apis.owner}/${api.name}/${api.version}</inputSpec>
							
							<auth>Authorization: ${swaggerhub.apis.token}</auth>
							
							<language>openapi-yaml</language>
							
							<output>${codegen.directory}</output>

							<configOptions>
								
								<outputFile>${app.swagger.file}</outputFile>
							</configOptions>
						</configuration>
					</execution>`

**Comment by michaelmelodysmartbear** [comment]
@olensmar I have to agree with @scorpioa on this one. As part of our backend services, we use the swaggerhub-maven-plugin in conjunction with the swagger-codegen-maven-plugin. Firstly to download the unresolved definition and then to use that for generation.

This plugin is in essence a wrapper of the SwaggerHub API. At this time, there is no API available to download resolved definitions.

I'll close this issue unless there are any objections?

**Comment by olensmar** [comment]
thanks @michaelmelodysmartbear - i'm fine with that workaround for now - although I think we should revisit this if we start seeing external usage of the plugin for other purposes than running the downloaded spec through codegen!

**Comment by michaelmelodysmartbear** [comment]
@olensmar I've came to the realization that you were right in creating this issue! Currently facing my own issues with unresolved definitions...
